### PR TITLE
Set up Associations

### PR DESCRIPTION
**Pull request by AbelG101**
## Changes
- Added 3 classes:
    - A Classroom class with a one-to-many relationship with the Student class
    - A Rental class as a linking class between Person and Book class 
        - stores the date, person, and book that is rented.
    - A Book class with a one-to-many relationship with the Rental class

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require_relative './student'
+
+class Classroom
+  attr_accessor :label
+
```
Comment (brytebee) [inline classroom.rb:6]: - [ ] **Required:** Kindly add a getter for the students' instance variable. Consider the code snippet below:
`attr_reader :students`

File: student.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ class Student < Person
   def initialize(classroom, age, name = 'unknown', parent_permmission: true)
     super(age, name, parent_permmission)
     @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
```
Comment (brytebee) [inline student.rb:13]: - [ ] **Required:** Kindly update this class to include the classroom setter and an accessor to get it. Consider the code below:

```ruby
 def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @tittle = title
+    @author = author
+    @rentals = []
+  end
```
Comment (brytebee) [inline book.rb:9]: - [ ] **Required:** This class should contain a method to add rentals. This class returns an instance of the Rental class.

File: person.rb [hunk 4]
```diff
@@ -7,6 +7,7 @@ def initialize(age, name = 'unknown', parent_permmission: true)
     @name = name
     @age = age
     @parent_permmission = parent_permmission
+    @rentals = []
   end
```
Comment (brytebee) [inline person.rb:12]: - [ ] **Required:** The Person class should also have a method to add rentals which return an instance of the Rental class.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @AbelG101 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great schema design.✔️
- The README looks professional.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @AbelG101 ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**